src/components/views/dialogs/DMInviteDialog.js

```diff
@@ -31,18 +31,46 @@ import {getHttpUriForMxc} from "matrix-js-sdk/lib/content-repo";
 const INITIAL_ROOMS_SHOWN = 3; // Number of rooms to show at first
 const INCREMENT_ROOMS_SHOWN = 5; // Number of rooms to add when 'show more' is clicked
 
-class DirectoryMember {
+// This is the interface that is expected by various components in this file. It is a bit
+// awkward because it also matches the RoomMember class from the js-sdk with some extra support
+// for 3PIDs/email addresses.
+//
+// Dev note: In order to allow us to compile the app correctly, this needs to be a class
```

Comment:
It doesn't sound like we actually use Flow at build time, so this comment may be incorrect. I don't see anything wrong with the code itself, but it would nice for at least this comment to reflect reality.

Reply:
I'm failing to follow any of the discussion about it in #riot-dev, but it definitely blows up webpack if any of this changes. It might not be flow, but it sure is something related to types. Even if the comment is a mild lie, I'd be tempted to keep it as a reminder that we should be using TypeScript instead of Flow/whatever this is.

Reply:
> even though Flow supports interfaces. It just means that we "extend" rather than "implement"

Flow might support it but currently all we have is babel set-up to strip flow-typing, we don't run through the flow-bin cli to do any other flow magic such as type checking.

Running flow type checking on our code base results in 57 errors such as:

```
Error ----------------------------------------------------------------------------------------- src/BasePlatform.js:53:9

Cannot assign `count` to `this.notificationCount` because property `notificationCount` is missing in `BasePlatform` [1].

   src/BasePlatform.js:53:9
   53|         this.notificationCount = count;
               ^^^^^^^^^^^^^^^^^^^^^^

References:
   src/BasePlatform.js:30:22
   30| export default class BasePlatform {
                            ^^^^^^^^^^^^ [1]
```

Reply:
@t3chguy ... so does the comment need changing? I'm still failing to parse anything you're saying.

Reply:
I guess so

> even though Flow supports interfaces. It just means that we "extend" rather than "implement"

is completely irrelevant, given we don't *actually* use Flow for anything other than type-checking in compatible IDEs, so we don't get this magical feature which it may support.

Reply:
I guess the comment could just be "XXX: Once we have Typescript support change this to an interface which other things implement instead of extending"